include/pulsar/Message.h

```diff
@@ -188,8 +188,7 @@ class PULSAR_PUBLIC Message {
     MessageImplPtr impl_;
 
     Message(MessageImplPtr& impl);
-    Message(const proto::CommandMessage& msg, proto::MessageMetadata& data, SharedBuffer& payload,
-            int32_t partition);
+    Message(const MessageId& messageId, proto::MessageMetadata& metadata, SharedBuffer& payload);
```

Comment:
Just have a thought, not a requested change and this PR needs no change about this comment. I think the constructor of `Message` should have only one private constructor.

```c++
Message(const MessageImplPtr& impl);
```

The other constructors should be moved to `MessageImpl`. After that, there is no need to expose the classes (even if as the forward declarations) in the header.

Reply:
Agree. We should make the constructor clearer and make this existing implementation out of the constructor.